### PR TITLE
validations, model specs, and n+1 problems

### PR DESCRIPTION
**Pull request by RolandM99**
### To be implemented: 

- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.
- Add flash messages in the create actions in all your controllers.
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+  describe 'validates' do
+    subject { User.new(name: 'Tom', photo: 'Tom.png', bio: 'Teacher from Mexico.') }
+
+    before { subject.save }
+
+    it 'should have a name' do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+
```
Comment (SajjadAhmad14) [inline spec/models/user_spec.rb:13]: Please add a unit test for `#recent_posts` of the User model.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+    # it 'return the three latest posts for a user' do
```
Comment (SajjadAhmad14) [inline spec/models/user_spec.rb]: Please remove commented code as it can make the whole code unclean.

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
+
+  it 'check if the title is not blank' do
+    subject.title = ''
+    expect(subject).to_not be_valid
+  end
+
+  it 'return when there is no comments yet for a post' do
+    expect(Comment.first).to eql nil
+  end
+
```
Comment (SajjadAhmad14) [inline spec/models/post_spec.rb:17]: Please add unit tests for `#update_posts_counter` and `#recent_comments` methods of Post Model.

File: app/models/post.rb [hunk 4]
```diff
@@ -1,4 +1,6 @@
 class Post < ApplicationRecord
+  validates :comment_counter, :likes_counter, numericality: { greater_than_or_equal_to: 0 }
+  validates :title, presence: true, length: { maximum: 250 }
   belongs_to :author, class_name: 'User'
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
```
Comment (SajjadAhmad14) [inline app/models/post.rb:6]: Please use flash messages in `comments#create` action for successful and not successful path and render it in views.

Comment (SajjadAhmad14) [inline app/models/post.rb:6]: Please solve the `N+1` problem in the `index` action of the `Posts` controller when fetching all posts and their comments for a user. You may learn about the N+1 problem [here](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh#:~:text=What%20is%20the%20N%2B1,the%20result%20of%20that%20data.).

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Roland, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🎉🍾✔**
Hello @RolandM99 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**